### PR TITLE
RI-80 Add ironic pre config to bootstrap-aio.yml

### PR DESCRIPTION
**Pull request by cjloader**
We removed the pre config playbook from the tree but instead
triggers the tasks inside bootstrap-aio.yml.

Issue: [RI-80](https://rpc-openstack.atlassian.net/browse/RI-80)

### CODE REVIEW

**Comment by cjloader** [comment]
recheck_cit_xenial_swift_deploy

File: scripts/bootstrap-aio.yml [hunk 1]
```diff
@@ -272,3 +272,98 @@
...
+          auto br-tftp
+          iface br-tftp inet static
+              bridge_stp off
+              bridge_waitport 10
+              bridge_fd 0
+              offload-sg off
+              bridge_ports none
+              address 172.29.232.100
+              netmask 255.255.252.0
+      when: "{{ lookup('env', 'DEPLOY_IRONIC') == 'yes' }}"
```
Comment (major) [inline scripts/bootstrap-aio.yml]: Ansible gets upset if you put jinja2 delimiters in `when:` clauses. You might want to put something like this up above where you see `DEPLOY_SWIFT`:

```yaml
rpco_deploy_ironic: "{{ lookup('env', 'DEPLOY_IRONIC') }}"
```

And then use `when: rpco_deploy_ironic == 'yes'` or `when: rpco_deploy_ironic | bool` in your tasks.

File: scripts/bootstrap-aio.yml [hunk 2]
```diff
@@ -272,3 +272,98 @@
...
+
+    - name: Start the ironic network interfaces
+      command: "ifup {{ item }}"
+      when: "{{ lookup('env', 'DEPLOY_IRONIC') == 'yes' }}"
```
Comment (major) [inline scripts/bootstrap-aio.yml]: See comment about jinja2 delimiters in `when` above.

File: scripts/bootstrap-aio.yml [hunk 3]
```diff
@@ -272,3 +272,98 @@
...
+      config_template:
+        src: "/etc/openstack_deploy/openstack_user_config.yml"
+        dest: "/etc/openstack_deploy/openstack_user_config.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_network_config }}"
+        config_type: "yaml"
+        list_extend: true
+      when: "{{ lookup('env', 'DEPLOY_IRONIC') == 'yes' }}"
```
Comment (major) [inline scripts/bootstrap-aio.yml]: See comment about jinja2 delimiters in `when` above.

File: scripts/bootstrap-aio.yml [hunk 4]
```diff
@@ -272,3 +272,98 @@
...
+        block: |
+          neutron_dhcp_config:
+            dhcp-option-force: "26,1500"
+            dhcp-ignore: "tag:!known"
+            log-facility: "/var/log/neutron/dnsmasq.log"
+          nova_scheduler_use_baremetal_filters: False
+          ironic_openstack_driver_list:
+            -  agent_ipmitool
+            -  fake
+      when: "{{ lookup('env', 'DEPLOY_IRONIC') == 'yes' }}"
```
Comment (major) [inline scripts/bootstrap-aio.yml]: See comment about jinja2 delimiters in `when` above.

File: scripts/bootstrap-aio.yml [hunk 5]
```diff
@@ -272,3 +272,99 @@
...
+            dhcp-ignore: "tag:!known"
+            log-facility: "/var/log/neutron/dnsmasq.log"
+          nova_scheduler_use_baremetal_filters: False
+          ironic_openstack_driver_list:
+            -  agent_ipmitool
+            -  fake
+      when: rpco_deploy_ironic | bool
+
+  vars:
+    rpco_deploy_swift: "{{ lookup('env', 'DEPLOY_IRONIC') }}"
```
Comment (major) [inline scripts/bootstrap-aio.yml]: swift to ironic here. :)

**Pull request merged**